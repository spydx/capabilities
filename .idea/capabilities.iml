<?xml version="1.0" encoding="UTF-8"?>
<module type="CPP_MODULE" version="4">
  <component name="NewModuleRootManager">
    <content url="file://$MODULE_DIR$">
      <sourceFolder url="file://$MODULE_DIR$/capabilities-derive/src" isTestSource="false" />
      <sourceFolder url="file://$MODULE_DIR$/src" isTestSource="false" />
      <sourceFolder url="file://$MODULE_DIR$/target/debug/build/async-trait-5b223cdf2dc4e051/out" isTestSource="false" />
      <sourceFolder url="file://$MODULE_DIR$/target/debug/build/proc-macro2-cecd7ae0bcf8d632/out" isTestSource="false" />
      <sourceFolder url="file://$MODULE_DIR$/target/debug/build/syn-12d88766fd7433d1/out" isTestSource="false" />
      <sourceFolder url="file://$MODULE_DIR$/tests" isTestSource="true" />
      <excludeFolder url="file://$MODULE_DIR$/target" />
    </content>
    <orderEntry type="inheritedJdk" />
    <orderEntry type="sourceFolder" forTests="false" />
  </component>
</module>