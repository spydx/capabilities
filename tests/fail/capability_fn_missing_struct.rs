use capabilities::SqliteDb;
use capabilities_derive::capabilities;
use capabilities_derive::capabilty;
use capabilities_derive::service;
use capabilities::*;


struct Orders {
    id: i32,
    name: String,
}

#[service(SqliteDb)]
#[tokio::main]
async fn main() -> Result<(), std::io::Error> {
    let connection_string = "sqlite::memory:".to_string();
    let _pool = CapService::build(connection_string)
        .await
        .expect("Failed to create database");
    Ok(())
}


#[capabilty(Orders, Read)]
fn get_order() -> Result<Orders, CapServiceErrro> {
    Ok(Orders {
        id: 1,
        name: String::from("lkajsdlkf"),
    })
}