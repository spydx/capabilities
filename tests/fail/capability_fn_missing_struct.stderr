error[E0432]: unresolved import `capabilities_derive::capabilty`
 --> tests/fail/capability_fn_missing_struct.rs:3:5
  |
3 | use capabilities_derive::capabilty;
  |     ^^^^^^^^^^^^^^^^^^^^^---------
  |     |                    |
  |     |                    help: a similar name exists in the module: `capability`
  |     no `capabilty` in the root

error: cannot determine resolution for the attribute macro `capabilty`
  --> tests/fail/capability_fn_missing_struct.rs:24:3
   |
24 | #[capabilty(Orders, Read)]
   |   ^^^^^^^^^
   |
   = note: import resolution is stuck, try simplifying macro imports

error[E0412]: cannot find type `CapServiceErrro` in this scope
  --> tests/fail/capability_fn_missing_struct.rs:25:34
   |
13 | #[service(SqliteDb)]
   | -------------------- similarly named struct `CapServiceError` defined here
...
25 | fn get_order() -> Result<Orders, CapServiceErrro> {
   |                                  ^^^^^^^^^^^^^^^
   |
help: a struct with a similar name exists
   |
25 | fn get_order() -> Result<Orders, CapServiceError> {
   |                                  ~~~~~~~~~~~~~~~
help: you might be missing a type parameter
   |
25 | fn get_order<CapServiceErrro>() -> Result<Orders, CapServiceErrro> {
   |             +++++++++++++++++

warning: unused import: `capabilities_derive::capabilities`
 --> tests/fail/capability_fn_missing_struct.rs:2:5
  |
2 | use capabilities_derive::capabilities;
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default
